src/vmm/src/device_manager/persist.rs

```diff
@@ -420,16 +418,17 @@ mod tests {
                 network_interface,
             );
             // Add a vsock device.
-            let orig_tmp_sock_file = TempSockFile::new(TempFile::new().unwrap());
+            let mut orig_tmp_sock_file = TempFile::new().unwrap();
+            orig_tmp_sock_file.remove();
             let vsock_dev_id = "vsock";
             let vsock_config = VsockDeviceConfig {
                 vsock_id: vsock_dev_id.to_string(),
                 guest_cid: 3,
-                uds_path: orig_tmp_sock_file.path().clone(),
+                uds_path: orig_tmp_sock_file.as_path().to_str().unwrap().to_string(),
             };
             insert_vsock_device(&mut vmm, &mut cmdline, &mut event_manager, vsock_config);
             // This will be used by the restored device and will cleanup the UDS when test ends.
-            _tmp_sock_file = orig_tmp_sock_file.clone();
+            _tmp_sock_file = orig_tmp_sock_file.as_path().to_str().unwrap().to_string();
```

Comment:
In the old version of the code, this object would do `fs::remove_file` on its `drop`. With your change this is no longer the case. You are only duplicating the path here and not the logic to auto clean-up that path when it goes out of scope.

I think what you need to do in this case is:

create tmpfile at the beginning of the test and call `remove()` on it so there is nothing at that path, then after the code-block `// Set up a vmm with one of each device, and get the serialized DeviceStates` call `remove()` on it to again delete what the code-block created at that path; at the end of the test it will automatically call `remove()` again.

```rust
let mut tmp_sock_file = TempFile::new().unwrap();
tmp_sock_file.remove();
// Set up a vmm with one of each device, and get the serialized DeviceStates.
let original_mmio_device_manager = {
    ...
};
tmp_sock_file.remove();
// rest of the test here
...
```

Reply:
Thank you for the feedback. I updated the code to reflect your suggestion.